solr/solrj/src/java/org/apache/solr/common/cloud/ReplicaCount.java

```diff
@@ -61,6 +122,11 @@ public boolean contains(Replica.Type type) {
     return countByType.containsKey(type);
   }
 
+  /** Returns the replica types for which a number of replicas was explicitely defined. */
+  public Set<Replica.Type> keySet() {
```

Comment:
Do we still need this method if this class implements a method to directly count the number of leader replicas or if there's at least one leader replica? (see comment on `count()` below)
(doing the review in GitHub not in the IDE so can't easily find all callers)

Reply:
It's used every time I need to iterate over replica types explicitly defined in this class. A similar behaviour could be obtained by iterating over all `Replica.Type#values` and calling `ReplicaCount#contains`, but it seemed more straightforward this way. We could also have a `ReplicaCoubt#forEach` method... So many ways to implement this!